py/server/deephaven/experimental/iceberg.py

```diff
@@ -265,57 +266,90 @@ def adapter(
             `nessie`, `jdbc`.
     - `uri` - the URI of the catalog
 
+    If the user is responsible for connecting to an AWS catalog or S3-compatible API for the warehouse, setting
+    s3_instructions is the recommended way of configuring these details, as it allows Deephaven to own the creation of
+    the Iceberg-internal AWS / S3 client. When set, these instructions should also be included as part of
+    IcebergInstructions data_instructions. Note: some REST Catalog implementations may be responsible for providing the
```

Comment:
I'm not sure what you mean. (Note: in my most recent commit, this will automatically be inherited correctly if the user doesn't set it on their IcebergInstructions.)

Reply:
Yea, that's what I was referring, that if a user doesn't provide data_instructions, would they be automatically backfilled from the catalog adapter. I thought Larry was working on that part, but good to know that you have added it in this commit only.